biz.aQute.bndlib/src/aQute/bnd/build/Project.java

```diff
@@ -2957,4 +2959,66 @@ public void setPackageInfo(String packageName, Version newVersion) throws Except
 	public Version getPackageInfo(String packageName) throws Exception {
 		return packageInfo.getPackageInfo(packageName);
 	}
+
+	/**
+	 * Actions to perform before a full workspace release. This is executed for
+	 * projects that describe the distribution
+	 */
+
+	public void preRelease() {
+		for (ReleaseBracketingPlugin rp : getWorkspace().getPlugins(ReleaseBracketingPlugin.class)) {
+			rp.begin(this);
+		}
+	}
+
+	/**
+	 * Actions to perform after a full workspace release. This is executed for
+	 * projects that describe the distribution
+	 */
+
+	public void postRelease() {
+		for (ReleaseBracketingPlugin rp : getWorkspace().getPlugins(ReleaseBracketingPlugin.class)) {
+			rp.end(this);
+		}
+	}
+
+	/**
+	 * Copy a repository to another repository
+	 *
+	 * @throws Exception
+	 */
+
+	public void copy(RepositoryPlugin source, String filter, RepositoryPlugin destination) throws Exception {
+		copy(source, filter == null ? null : new Instructions(filter), destination);
+	}
+	public void copy(RepositoryPlugin source, Instructions filter, RepositoryPlugin destination) throws Exception {
+
+		assert source != null;
+		assert destination != null;
+
+		logger.info("copy from repo {} to {} with filter {}", source, destination, filter);
+
+		for (String bsn : source.list(null)) {
+			for (Version version : source.versions(bsn)) {
+				if (filter == null || filter.matches(bsn)) {
+					logger.info("copy {}:{}", bsn, version);
+					File file = source.get(bsn, version, null);
+					if (file.getName().endsWith(".jar")) {
+						try (InputStream in = new FileInputStream(file)) {
```

Comment:
Don't use `new FileInputStream`. I went through the code base to remove that usage. Please use `IO.stream` instead.